lib/vdc/Grid.cpp

```diff
@@ -22,6 +22,45 @@
 using namespace std;
 using namespace VAPoR;
 
+namespace {
+
+bool interpolate_point_on_node(const std::array<float, 4> &verts, double xwgt, double ywgt, float mv, float &v)
+{
+    if (xwgt == 1.0 && ywgt == 1.0) {
+        v = verts[0];
+        return (true);
+    }
+    if (xwgt == 0.0 && ywgt == 1.0) {
+        v = verts[1];
+        return (true);
+    }
+    if (xwgt == 1.0 && ywgt == 0.0) {
+        v = verts[2];
+        return (true);
+    }
+    if (xwgt == 0.0 && ywgt == 0.0) {
+        v = verts[3];
+        return (true);
+    }
+
+    return (mv);
+}
+
+bool interpolate_point_on_edge(const std::array<float, 4> &verts, double xwgt, double ywgt, float mv, float &v)
+{
+    if (ywgt == 1.0 && xwgt > 0.0 && xwgt < 1.0 && verts[0] != mv && verts[1] != mv) { v = (verts[0] * xwgt) + (verts[1] * (1.0 - xwgt)); }
+
+    if (xwgt == 0.0 && ywgt > 0.0 && ywgt < 1.0 && verts[1] != mv && verts[3] != mv) { v = (verts[1] * ywgt) + (verts[3] * (1.0 - ywgt)); }
+
+    if (ywgt == 0.0 && xwgt > 0.0 && xwgt < 1.0 && verts[2] != mv && verts[3] != mv) { v = (verts[2] * xwgt) + (verts[3] * (1.0 - xwgt)); }
+
+    if (xwgt == 1.0 && ywgt > 0.0 && ywgt < 1.0 && verts[0] != mv && verts[2] != mv) { v = (verts[0] * ywgt) + (verts[2] * (1.0 - ywgt)); }
+
+    return (mv);
```

Comment:
Oops. Good catch. We apparently don't have a test case that hits this code. Once again it would be great to have a unit test framework to catch these things.